runtime/bcutil/ClassFileWriter.hpp

```diff
@@ -350,14 +350,31 @@ class ClassFileWriter {
 			_anonClassName = J9ROMCLASS_CLASSNAME(_romClass);
 			U_16 anonNameLength = J9UTF8_LENGTH(_anonClassName);
 			U_16 originalNameLength = anonNameLength - ROM_ADDRESS_LENGTH - 1;
+			U_8 *anonClassNameData = J9UTF8_DATA(_anonClassName);
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+#define J9_INJECTED_INVOKER_CLASSNAME "InjectedInvoker"
+			/* ROM class format: <HOST_NAME>/InjectedInvoker/<ROM_ADDRESS_LENGTH>.
+			 * Search for InjectedInvoker in _anonClassName using the above format.
+			 * If found, reset the class name to "InjectedInvoker" in the class file.
+			 */
+			IDATA startIndex = anonNameLength - LITERAL_STRLEN(J9_INJECTED_INVOKER_CLASSNAME) - ROM_ADDRESS_LENGTH - 1;
+			U_8 *start = anonClassNameData + startIndex;
+			if ((startIndex >= 0)
+			&& (0 == memcmp(start, J9_INJECTED_INVOKER_CLASSNAME, LITERAL_STRLEN(J9_INJECTED_INVOKER_CLASSNAME)))
+			) {
+				originalNameLength = LITERAL_STRLEN(J9_INJECTED_INVOKER_CLASSNAME);
+				anonClassNameData = (U_8 *)J9_INJECTED_INVOKER_CLASSNAME;
+			}
```

Comment:
you can use `J9_DECLARE_CONSTANT_UTF8` since you know that in this case the name will always be `InjectedInvoker` and that way you can avoid the allocation

Reply:
`J9_DECLARE_CONSTANT_UTF8` puts the string in static memory, so you need to remember to not free the `anonClassNameData` in that case

Reply:
`J9_DECLARE_CONSTANT_UTF8` puts the string in static memory, so you need to remember to not free